openfe/protocols/openmm_utils/settings_validation.py

```diff
@@ -0,0 +1,83 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+"""
+Reusable utility methods to validate input settings to OpenMM-based alchemical
+Protocols.
+"""
+import warnings
+from typing import Dict, List, Tuple
+from openff.units import unit
+from gufe import (
+    Component, ChemicalSystem, SolventComponent, ProteinComponent
+)
+
+
+def validate_timestep(hmass: float, timestep: unit.Quantity):
+    """
+    Check that the input timestep is suitable for the given hydrogen
+    mass.
+
+    Parameters
+    ----------
+    hmass : float
+      The target hydrogen mass (assumed units of amu).
+    timestep : unit.Quantity
+      The integration time step.
+
+
+    Raises
+    ------
+    ValueError
+      If the hydrogen mass is less than 3 amu and the timestep is
+      greater than 2 fs.
+    """
+    if hmass < 3.0:
+        if timestep > 2.0 * unit.femtoseconds:
+            errmsg = f"timestep {timestep} too large for hydrogen mass {hmass}"
+            raise ValueError(errmsg)
+
+
+def get_simsteps(equil_length: unit.Quantity, prod_length: unit.Quantity,
+                 timestep: unit.Quantity, mc_steps: int) -> Tuple[int, int]:
+    """
+    Gets and validates the number of equilibration and production steps.
+
+    Parameters
+    ----------
+    equil_length : unit.Quantity
+      Simulation equilibration length.
+    prod_length : unit.Quantity
+      Simulation production length.
+    timestep : unit.Quantity
+      Integration timestep.
+    mc_steps : int
+      Number of integration timesteps between MCMC moves.
+
+    Returns
+    -------
+    equil_steps : int
+      The number of equilibration timesteps.
+    prod_steps : int
+      The number of production timesteps.
+    """
+
+    equil_time = round(equil_length.to('attosecond').m)
+    prod_time = round(prod_length.to('attosecond').m)
+    ts = round(timestep.to('attosecond').m)
+
+    equil_steps, mod = divmod(equil_time, ts)
+    if mod != 0:
+        raise ValueError("Equilibration time not divisible by timestep")
```

Comment:
happy to punt this to a later issue, but here and elsewhere it sounds saner to instead extend equil_time to be a multiple of ts

Reply:
This isn't how you do adaptive length.  Dw about it, when we cross that bridge I know what to do.